### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by Lockeriano**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -12,15 +12,15 @@ gem 'pg', '~> 0.18'
...
+gem 'rubocop', '0.48.1'
 gem 'rubocop-rspec', '1.8.0'
 gem 'sass-rails', '~> 5.0'
 gem 'turbolinks', '~> 5'
 gem 'uglifier', '>= 1.3.0'
 
 group :development, :test do
   gem 'byebug', platforms: %i(mri mingw x64_mingw)
-  gem 'capybara', '~> 2.13'
+  gem 'capybara', '~> 2.16', '>= 2.16.1'
```
Comment (goozzik) [inline Gemfile]: `~> 2.16` is enough here. What It basically does is that it takes the most recent `2.16.x` version, so let's say that we have such versions available:

- `2.16.1`
- `2.16.2`

It will take the second one.

File: spec/features/recipes/welcome_page_spec.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+feature 'User visits' do
+  scenario 'valid page' do
+    visit('/')
+    expect(page).to have_content "We're gonna have some fun creating this app, old pal"
+  end
+end
```
Comment (goozzik) [inline spec/features/recipes/welcome_page_spec.rb]: 1. Please prepare proper description for `feature`. It should describe in simple words the feature we are testing.
2. Please use word `it` instead of `scenario`. Because when you read it you can notice that if feels more natural (look example at the bottom)
3. Please prepare proper description for `it`. It should describe in simple words the environment changes that happen in the given `scenario` block.
4. Keep all the environment changes outside of the `it` block whenever it is possible. The exception would be case where we have more steps in the middle:
  ```ruby
    before do
      visit('/')  
    end

    it 'redirects user to test page' do
      expect(page).to have_content "We're gonna have some fun creating this app, old pal"

      click_link('go to test page') # Good candidate to move out to `before` but since it happens in the middle we keep it here.
      expect(page).to have_content "We're on the test page!"
    end
  ```

So to summarize it should look like this:

```ruby
feature 'User visits welcome page' do
  before { visit('/') }

  it 'shows user welcome message' do
    expect(page).to have_content "We're gonna have some fun creating this app, old pal"
  end
end
```

File: spec/features/recipes/welcome_page_spec.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
```
Comment (goozzik) [inline spec/features/recipes/welcome_page_spec.rb]: Please DRY this up. We can have `.rspec` (read the docs) file in the root with this configuration.

File: .rspec [hunk 4]
```diff
@@ -1 +1,2 @@
 --require spec_helper
```
Comment (goozzik) [inline .rspec]: We can now get rid of this since this is already required by `rails_helper`.

File: Gemfile [hunk 5]
```diff
@@ -12,15 +12,15 @@ gem 'pg', '~> 0.18'
...
+gem 'rubocop', '0.48.1'
 gem 'rubocop-rspec', '1.8.0'
 gem 'sass-rails', '~> 5.0'
 gem 'turbolinks', '~> 5'
 gem 'uglifier', '>= 1.3.0'
 
 group :development, :test do
   gem 'byebug', platforms: %i(mri mingw x64_mingw)
-  gem 'capybara', '~> 2.13'
+  gem 'capybara', '~> 2.16'
```
Comment (goozzik) [inline Gemfile]: Let's move it to `test` group, no need to have it in `development`.

File: Gemfile [hunk 6]
```diff
@@ -12,15 +12,15 @@ gem 'pg', '~> 0.18'
...
   gem 'factory_bot_rails'
   gem 'rspec-rails', '~> 3.6'
   gem 'selenium-webdriver'
```
Comment (goozzik) [inline Gemfile]: Let's move it to `test` group, no need to have it in `development`.

File: spec/rails_helper.rb [hunk 7]
```diff
@@ -4,10 +4,10 @@
 require File.expand_path('../../config/environment', __FILE__)
 # Prevent database truncation if the environment is production
 abort('The Rails environment is running in production mode!') if Rails.env.production?
-require 'rspec/rails'
```
Comment (goozzik) [inline spec/rails_helper.rb]: actually this one should stay here to keep the separation :)

File: Gemfile [hunk 8]
```diff
@@ -20,7 +20,6 @@ gem 'uglifier', '>= 1.3.0'
 
 group :development, :test do
   gem 'byebug', platforms: %i(mri mingw x64_mingw)
-  gem 'capybara', '~> 2.13'
   gem 'factory_bot_rails'
   gem 'rspec-rails', '~> 3.6'
   gem 'selenium-webdriver'
```
Comment (goozzik) [inline Gemfile]: Please move this to `test` before merge.

File: Gemfile [hunk 9]
```diff
@@ -33,4 +32,9 @@ group :development do
   gem 'web-console', '>= 3.3.0'
 end
 
+group :test do
+  gem 'capybara', '~> 2.16'
+  gem 'database_cleaner'
```
Comment (goozzik) [inline Gemfile]: Please lock the version `~> 1.6.2`
